src/main/java/org/jetbrains/bsp/bazel/server/bsp/BazelBspServerBuildManager.java

```diff
@@ -181,22 +191,22 @@ private JvmBuildTarget getJVMBuildTarget() {
 
   public Either<ResponseError, CompileResult> buildTargetsWithBep(
       List<BuildTargetIdentifier> targets, List<String> extraFlags) {
-    List<String> args = Lists.newArrayList(Constants.BAZEL_BUILD_COMMAND);
-    args.addAll(targets.stream().map(BuildTargetIdentifier::getUri).collect(Collectors.toList()));
-    args.addAll(extraFlags);
-    int exitCode = -1;
+    List<String> bazelTargets =
+        targets.stream().map(BuildTargetIdentifier::getUri).collect(Collectors.toList());
+
+    StatusCode exitCode = StatusCode.ERROR;
 
     final Map<String, String> diagnosticsProtosLocations =
         bepServer.getDiagnosticsProtosLocations();
-    Build.QueryResult queryResult =
-        queryResolver.getQuery(
-            "query",
-            "--output=proto",
-            "("
-                + targets.stream()
-                    .map(BuildTargetIdentifier::getUri)
-                    .collect(Collectors.joining("+"))
-                + ")");
+    BazelProcessResult bazelProcessResult =
+        bazelRunner
+            .commandBuilder()
+            .query()
+            .withFlag(BazelRunnerFlag.OUTPUT_PROTO)
+            .withTargets(bazelTargets)
+            .executeBazelCommand();
+
+    Build.QueryResult queryResult = QueryResolver.getQueryResultForProcess(bazelProcessResult);
```

Comment:
Why is ``QueryResolver`` still around only with this one method that is just a wrapped _try-catch_ over ``Build.QueryResult::parseFrom``? I would remove it.

Reply:
it needs to be changed, but not in this PR, in general flow should be updated (including resolvers)

Reply:
Okay, I thought ``QueryResolver`` was larger and your PR disassembled it more.